ocaml/search.ml

```diff
@@ -0,0 +1,43 @@
+let fold_lines f init file =
+  let input = open_in file in
+  let rec fold state =
+    match input_line input with
+    | line -> fold (f state line)
+    | exception _ -> close_in input; state in
+  fold init
+
+open Batteries
+
+let knicks_re =
+  Re_pcre.re "^.*\t(.*)\t.*\t.*knicks.*$" |> Re.no_case |> Re.whole_string |> Re.compile
```

Comment:
Indeed; note that many languages test for "knick" in the whole line because it's more convenient, code-wise, than checking in the message text only. I don't think this affects efficiency, but it means that they would have completely incorrect results if a neighbourhood had "knicks" in its name.